### PR TITLE
Add more tests for Unicode case-insensitivity in regexes.

### PR DESCRIPTION
**Pull request by sjrd**
This was woefully under-tested.

### CODE REVIEW

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala [hunk 1]
```diff
@@ -1978,6 +1978,37 @@ class RegexEngineTest  {
...
+    // 0175-0182 (contains 017F which folds to 's')
+    if (!executingInJVM) { // looks like a JVM bug
+      assertMatches(ranges, "s")
+      assertMatches(ranges, "S")
+    }
+    assertMatches(ranges, "\u017F") // ſ LATIN SMALL LETTER LONG S
+    assertMatches(ranges, "\u0180") // neither mapped to nor from, but in the specified range
+    // 0540-0550
+    assertMatches(ranges, "\u0547") // in range
+    assertMatches(ranges, "\u0577") // mapped from 0577
```
Comment (gzm0) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala]: Wrong code point in comment?

Also: When you write "mapped from X" do you mean the code point in the test is folded to from that value?

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala [hunk 2]
```diff
@@ -1978,6 +1978,37 @@ class RegexEngineTest  {
...
+
+    val ranges = compile("[g-l\uFB00\u0175-\u0182\u0540-\u0550\u1F68-\u1F8E\u1FAA-\u1FAF\u2126]",
+        CaseInsensitive | UnicodeCase)
+    // g-l
+    assertMatches(ranges, "H")
+    assertMatches(ranges, "\u212A") // K KELVIN SIGN, maps to k
+    // FB00
+    assertMatches(ranges, "\uFB00") // ﬀ LATIN SMALL LIGATURE FF
+    // 0175-0182 (contains 017F which folds to 's')
+    if (!executingInJVM) { // looks like a JVM bug
```
Comment (gzm0) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala]: Were you able to find a reference for this? (or is this why I'm not requested as a reviewer on this PR yet and I'm prematurely reviewing? :P).

Comment (sjrd) [reply test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala]: No, I haven't been able to find a reference. The other way around does match (using `r-t` in the pattern, and matching `"\u017F"`, though, so it is most likely a bug. I should add that reverse thing as a test case to demonstrate that, actually.

I should probably file a bug to OpenJDK as well.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala (hunk 1 above)
Comment (sjrd) [reply test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala]: Oops, yes, the comment should say "0547".

> Also: When you write "mapped from X" do you mean the code point in the test is folded to from that value?

Yes, that's right. It might be clearer to write "0547 folds to 0577" in the comment?

**Comment by sjrd** [comment]
Also FYI this is in the context of https://github.com/scala-wasm/scala-wasm/pull/83.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala (hunk 1 above)
Comment (gzm0) [reply test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala]: Yes, I think that would be much clearer.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala (hunk 2 above)
Comment (sjrd) [reply test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexEngineTest.scala]: Bug filed; pending review.

**Comment by sjrd** [comment]
Can be reviewed again. I suggest we wait for the bug to be reviewed and added to the bug database before merging, so that we can directly include a link.

**Comment by sjrd** [comment]
Bug report accepted upstream. I inserted the link.

**Review by gzm0 — APPROVED** [review]
